### PR TITLE
Ration

### PR DESCRIPTION
**Pull request by katrina433**
Updates: 
calf_ration.py: implemented calf ration to return animal_intake and nutrient_requirements. 
pen.py: separated avg_calf_nutrient_rqmts from avg_nutrient_rqmts. 
simulation_engine.py, classes.py, animal_management.py, life_cycle.py, calf.py: passed down weather and time arguments to get the daily temperature that is needed in calf ration calculation. 
animal_management_animal.json and barnyard_animal.json: added wean_length and milk_type

### CODE REVIEW

**Review by Jacob8399 — COMMENTED** [review]
I looked through the changed files and everything seems to be in working order. dj_fields was merged yesterday so before I approve it you will need to update this branch with the new master and resolve the possible conflicts. After that, I'll check it again and then approve it.

**Review by Jacob8399 — APPROVED** [review]

**Review by tangswm — APPROVED** [review]
I've combed through all of the changes and they look good. Everything seems to be working well. I think these changes should be approved.

File: RUFAS/routines/animal/ration/calf_ration.py [hunk 1]
```diff
@@ -7,3 +9,182 @@ def optimize(feed, rqmts):
 
 def calculate_rqmts():
     return get_nutrient_rqmts()
+
+def calc_requirements(calf, temp, wean_day, wean_length, milk_type):
+    '''
```
Comment (manfei-L) [inline RUFAS/routines/animal/ration/calf_ration.py:10]: The names of these two functions seem not so distinguished. calculate_rqmts and calc_requirements.

Comment (katrina433) [reply RUFAS/routines/animal/ration/calf_ration.py:10]: calculated_rqmts() was the original hard coded ration, I can delete this function.

**Review by manfei-L — APPROVED** [review]
It looks good to me.

**Pull request merged**